src/Tgstation.Server.Host/Security/OAuth/InvisionCommunityOAuthValidator.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Net.Http;
+
+using Microsoft.Extensions.Logging;
+
+using Tgstation.Server.Api.Models;
+using Tgstation.Server.Host.Configuration;
+using Tgstation.Server.Host.System;
+
+namespace Tgstation.Server.Host.Security.OAuth
+{
+	/// <summary>
+	/// OAuth validator for Discord.
+	/// </summary>
+	sealed class InvisionCommunityOAuthValidator : GenericOAuthValidator
+		{
+		/// <inheritdoc />
+		public override OAuthProvider Provider => OAuthProvider.InvisionCommunity;
+
+		/// <inheritdoc />
+		protected override Uri TokenUrl => new Uri($"{BaseProtocolPath}/oauth/token/");
+
+		/// <inheritdoc />
+		protected override Uri UserInformationUrl => new Uri($"{BaseProtocolPath}/api/core/me");
+
+		/// <summary>
+		/// Base path to the server's OAuth endpoint.
+		/// </summary>
+		string BaseProtocolPath => $"{OAuthConfiguration.ServerUrl}";
```

Comment:
I can tell you copypasta'd the Keycloak provider, but this property is unecessary if it's just referencing the variable